bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/handler/LinkyHandler.java

```diff
@@ -211,8 +216,9 @@ private synchronized void updatePowerData() {
         if (isLinked(PEAK_POWER) || isLinked(PEAK_TIMESTAMP)) {
             cachedPowerData.getValue().ifPresentOrElse(values -> {
                 Aggregate days = values.aggregats.days;
-                updatekVAChannel(PEAK_POWER, days.datas.get(days.datas.size() - 1));
-                updateState(PEAK_TIMESTAMP, new DateTimeType(days.periodes.get(days.datas.size() - 1).dateDebut));
+                updatekVAChannel(PEAK_POWER, days.datas.get(days.datas.size() - 1).valeur);
+                updateState(PEAK_TIMESTAMP, new DateTimeType(
+                        days.datas.get(days.datas.size() - 1).dateDebut.atZone(this.timeZoneProvider.getTimeZone())));
```

Comment:
This was changed very recently, just before releasing 4.3. We don't need to pass a ZonedDateTipe; DateTimeType just needs an Instant as parameter. So you do not need TimeZoneProvider .

Reply:
Hello @lolodomo,

I'm not sure to understant. Do you say to use the DateTimeType constructor that take an instant, something like this.
 updateState(PEAK_TIMESTAMP,
                        new DateTimeType(days.datas.get(days.datas.size() - 1).dateDebut.toInstant(zoneOffset)));

The matter is that I still need to find a zoneOffset in this case.
The Enedis API return a LocalDateTime at offset of the current zone.
The DateTimeType if I undestand correctly want a universal time.

So somewhere we need to convert from local to universal using a timezone ?

Laurent.

Reply:
@jlaur is our specialist with date. Please advice.

Reply:
If `days.datas.get(days.datas.size() - 1).dateDebut` is a `LocalDateTime`, the implementation is already correct. If it is an `Instant` there is no need to convert to `ZonedDateTime`.

Reply:
> the implementation is already correct

@jlaur : you mean @lo92fr proposed code is correct ?
I am a little lost, why should we consider the timezone setup in server ?
Imagine that for a strange reason I setup timezone in OH server to UTC+5. I am convinced Enedis provides dates in France timezone, that is UTC+1.

Reply:
> @jlaur : you mean @lo92fr proposed code is correct ?

Yes, that's what I meant, but purely from a technical perspective, i.e. if you only have a `LocalDateTime`, you also need a `ZoneId` regardless of if you want an `Instant` or `ZonedDateTime`.

> I am a little lost, why should we consider the timezone setup in server  Imagine that for a strange reason I setup timezone in OH server to UTC+5. I am convinced Enedis provides dates in France timezone, that is UTC+1

Indeed, if the local datetime is from a specific server, you need to use the time-zone from that server. This is similar:

https://github.com/openhab/openhab-addons/blob/main/bundles%2Forg.openhab.binding.energidataservice%2Fsrc%2Fmain%2Fjava%2Forg%2Fopenhab%2Fbinding%2Fenergidataservice%2Finternal%2FEnergiDataServiceBindingConstants.java#L88-L90

For a local device, it would make sense to assume same time-zone as openHAB.